### ISSUE TITLE
Undefined index: extension at /var/www/owncloud/apps/files_versions/lib/storage.php#320

### ISSUE THREAD

**Issue by enoch85** [owncloud/core#14408]
Just wanted to inform you about an undefined index:

```
Error	PHP	Undefined index: extension at /var/www/owncloud/apps/files_versions/lib/storage.php#320	2015-02-20T08:37:19+00:00
```

ownCloud 8.0

**Comment by PVince81** [comment]
I tried creating versions for a file without extension, but couldn't make this warning appear (on stable8).
It must be something else.

**Comment by enoch85** [comment]
@PVince81 Thanks for trying.

**Comment by nickvergessen** [comment]
@enoch85 any idea how to reproduce this (maybe you can try to find which file was deleted/created/restored/whatever at that time from your access log?

**Comment by enoch85** [comment]
```
7 1528 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:37:08 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:37:09 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1539 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 82.44.188.2 - - [20/Feb/2015:08:37:10 +0100] "POST /index.php/heartbeat HTTP/1.1" 200 7271 "https://en0ch.se/index.php/s/jaBcGYB0BkjaVY7?path=%2FownCloud%20pre-installed%20%5BStable%208%5D%20%5BSSL%5D%20%5BUbuntu%20Server%2014.04%5D" "Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.111 Safari/537.36"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:37:11 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 7656 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:37:17 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 5862 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:37:23 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1485 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:37:24 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1752 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 141.101.64.161 - - [20/Feb/2015:08:37:31 +0100] "GET /owncloud_/pre-configured-owncloud-installaton/feedback/ HTTP/1.1" 200 12186 "https://www.en0ch.se/owncloud_/stress-test/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 141.101.64.161 - - [20/Feb/2015:08:37:32 +0100] "GET /wp-content/plugins/counterize/counterize.js.php?ver=4.1.1 HTTP/1.1" 200 4002 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/feedback/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 141.101.64.161 - - [20/Feb/2015:08:37:32 +0100] "GET /wp-content/plugins/counterize/counterize.css.php?ver=4.1.1 HTTP/1.1" 200 7326 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/feedback/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:37:33 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1528 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:37:38 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:37:39 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1539 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:37:41 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1605 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:37:47 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 5862 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 141.101.64.161 - - [20/Feb/2015:08:37:52 +0100] "GET /owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/ HTTP/1.1" 200 13477 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/feedback/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 141.101.64.161 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/plugins/counterize/counterize.css.php?ver=4.1.1 HTTP/1.1" 200 1583 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 173.245.53.113 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/uploads/2015/01/gateway.png HTTP/1.1" 304 5961 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 141.101.64.191 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/uploads/2015/01/interfaces-300x184.png HTTP/1.1" 304 5962 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 173.245.53.120 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/uploads/2015/01/tab-300x199.jpg HTTP/1.1" 304 5962 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 141.101.64.185 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/uploads/2015/01/player-300x270.png HTTP/1.1" 304 5963 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 141.101.64.161 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/plugins/counterize/counterize.js.php?ver=4.1.1 HTTP/1.1" 200 9745 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 173.245.53.68 - - [20/Feb/2015:08:37:52 +0100] "GET /wp-content/uploads/2015/01/Putty_settings-300x290.png HTTP/1.1" 304 5963 "https://www.en0ch.se/owncloud_/pre-configured-owncloud-installaton/complete-install-instructions/" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:37:54 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1752 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:37:55 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1485 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 193.107.225.37 - - [20/Feb/2015:08:37:56 +0100] "POST /index.php/heartbeat HTTP/1.1" 200 7271 "https://en0ch.se/index.php" "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.111 Safari/537.36"
en0ch.se:443 195.145.245.148 - - [20/Feb/2015:08:37:56 +0100] "POST /index.php/heartbeat HTTP/1.1" 200 7295 "https://en0ch.se/index.php/s/jaBcGYB0BkjaVY7?path=%2F" "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:30.0) Gecko/20100101 Firefox/30.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:38:03 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 7650 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:38:08 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:38:09 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1610 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:38:11 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1534 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:38:17 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 5862 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:38:24 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1752 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:38:27 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1485 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:38:33 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1528 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:38:38 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:38:39 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1539 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:38:41 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1534 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:38:47 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 5862 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:38:54 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1752 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:38:59 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1485 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:39:03 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1528 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:39:08 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:39:09 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1539 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:39:11 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1534 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:39:17 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 5862 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:39:24 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1752 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:39:31 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1485 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:39:34 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1386 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:39:38 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:39:39 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1539 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:39:41 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1534 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:39:47 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 5862 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:39:54 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1752 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:40:03 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1485 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:40:04 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1386 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - miralltest [20/Feb/2015:08:40:08 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1455 "-" "Mozilla/5.0 (Windows) mirall/1.8.0"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/2015:08:40:09 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1539 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 83.253.116.139 - horgle [20/Feb/2015:08:40:12 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1534 "-" "Mozilla/5.0 (Windows) mirall/1.7.1"
en0ch.se:443 192.168.1.1 - enoch [20/Feb/20
```

**Comment by DeepDiver1975** [comment]
Just had the same issue on 8.0.1 while deleting files via webdav using the desktop client

**Comment by DeepDiver1975** [comment]
{"reqId":"54fcf1ebeaae2c6b7799ec7fc8a34b7f","remoteAddr":"192.168.178.24","app":"PHP","message":"Undefined index: extension at \/home\/deepdiver\/Development\/ownCloud\/stable8\/apps\/files_versions\/lib\/storage.php#320","level":3,"time":"2015-03-05T15:14:11+00:00"}

**Comment by DeepDiver1975** [comment]
And I'm left with two stray version files ... awesome

**Comment by PVince81** [comment]
That could smell like concurrency, but not sure...

**Comment by iwurzer** [comment]
I have the same messages in our oc.log

Important: yesterday we had again some "mystic" file deletions. That means that a client, deleted some files/folders without an "user command". Short after this the same user - and only he/she" gets the entry within the log: {"reqId":"9c0cdcd6d6c2be2312cb203f5f6f5433"	remoteAddr:"xx.xx.xx.xx" app:"PHP"	message:"Undefined index: extension at \/var\/www\/owncloud\/apps\/files_versions\/lib\/storage.php#320"	level:3	time:"2015-06-23T17:17:55+02:00"}								

For us (company) this is a "no go" because there is nearly no trust anymore within owncloud, because you cannot be sure, that your data is safe...

**Comment by PVince81** [comment]
@iwurzer the mystic deletion could have been be https://github.com/owncloud/core/issues/13391 when renaming a huge folder.

@enoch85 also there were issues with the versions app where moving a folder didn't always move the versions with them. This should be fixed in 8.0.6 and 8.1.1

Please reopen if you still see these warnings in 8.1.1.
